### PR TITLE
scheduler, api(cdc): disable owner as the drain capture target.

### LINKED ISSUE #6042: schedulerV3 should not automatically balance table if any capture is stopping.
When there is a capture stopping, automatic balance table should be disabled, to prevent frequent unnecessary table scheduling.

### LINKED ISSUE #6043: owner should not be set as the drain capture target
as title

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #6042, #6043

### What is changed and how it works?

* owner is not allow be set as the drain capture target.
* `balanceScheduler` should not work if there is any capture stopping.

#### Tests 

 - Unit test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/scheduler/internal/tp/coordinator.go [hunk 1]
```diff
@@ -249,7 +250,8 @@ func (c *coordinator) poll(
 
 	// Generate schedule tasks based on the current status.
 	replications := c.replicationM.ReplicationSets()
-	allTasks := c.schedulerM.Schedule(checkpointTs, currentTables, aliveCaptures, replications)
+	allTasks := c.schedulerM.Schedule(checkpointTs, currentTables,
+		aliveCaptures, replications, c.captureM.HasCaptureStopping())
```
Comment (overvenus) [inline cdc/scheduler/internal/tp/coordinator.go:254]:
```suggestion
	allTasks := c.schedulerM.Schedule(
		checkpointTs, currentTables, aliveCaptures, replications, c.captureM.HasCaptureStopping())
```

File: cdc/scheduler/internal/tp/scheduler.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ type scheduler interface {
 		currentTables []model.TableID,
 		aliveCaptures map[model.CaptureID]*model.CaptureInfo,
 		replications map[model.TableID]*ReplicationSet,
-	) []*scheduleTask
+		hasCaptureStopping bool) []*scheduleTask
```
Comment (overvenus) [inline cdc/scheduler/internal/tp/scheduler.go:27]: What about embedding stopping in `aliveCaptures map[model.CaptureID]*CaptureStatus`?

File: cdc/api/v1/api.go [hunk 3]
```diff
@@ -743,6 +743,18 @@ func (h *OpenAPI) DrainCapture(c *gin.Context) {
...
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	if ownerInfo.ID == target {
+		_ = c.Error(cerror.ErrSchedulerRequestFailed.
+			GenWithStackByArgs("cannot drain the owner"))
+		return
+	}
```
Comment (overvenus) [inline cdc/api/v1/api.go:756]: Please add a test.

File: cdc/scheduler/internal/tp/scheduler.go (hunk 2 above)
Comment (3AceShowHand) [reply cdc/scheduler/internal/tp/scheduler.go:27]: I've tried it, and it works.

But `CaptureStatus` looks should be managed by `CaptureManager`. Scheduler does not need to know the existance of `CaptureStatus`.

Comment (3AceShowHand) [reply cdc/scheduler/internal/tp/scheduler.go:27]: The recommend format is adjusted by the linter.

File: cdc/api/v1/api_test.go [hunk 4]
```diff
@@ -435,6 +436,112 @@ func TestRebalanceTables(t *testing.T) {
...
+	data := model.DrainCaptureRequest{
+		CaptureID: captureInfo.ID,
+	}
+	b, err := json.Marshal(&data)
+	require.NoError(t, err)
+
+	body := bytes.NewReader(b)
+	api := testCase{
+		url:    "/api/v1/captures/drain",
+		method: "POST",
```
Comment (overvenus) [inline cdc/api/v1/api_test.go:460]: It occurs to me, we should use `PUT` as it idempotent [1]

\[1\]: https://restfulapi.net/rest-put-vs-post/

File: cdc/scheduler/internal/tp/scheduler_balance.go [hunk 5]
```diff
@@ -52,6 +53,14 @@ func (b *balanceScheduler) Schedule(
 	}
 	b.lastRebalanceTime = now
 
+	for _, capture := range captures {
+		if capture.State == CaptureStateStopping {
```
Comment (overvenus) [inline cdc/scheduler/internal/tp/scheduler_balance.go:57]: Could you add a test?

Comment (3AceShowHand) [reply cdc/scheduler/internal/tp/scheduler_balance.go:57]: it's covered.

**Review by overvenus — APPROVED** [review]
LGTM

**Review by asddongmen — APPROVED** [review]

**Pull request merged**